kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -82,7 +83,7 @@ public class TransactionImpl implements Transaction {
   private final Path dataPath;
   private final Path logPath;
   private final Protocol protocol;
-  private final SnapshotImpl readSnapshot;
+  private final Optional<SnapshotImpl> readSnapshot;
```

Comment:
nit: could we just pull the trigger and rename this to `readSnapshotOpt`?

Reply:
yep good call